### ISSUE TITLE
No longer maintained — Do you want to help?

### ISSUE THREAD

**Issue by paularmstrong** [paularmstrong/swig#628]
As you may have realized, Swig Templates is no longer maintained. I haven't had the time nor ambition to do anything to it for some time. I've never gotten the chance to even use my own project outside of just a statically generated personal site or two. 

Because of that, this project is no longer a priority for me. I'd rather spend my time outdoors and enjoying life outside of code than fix things on a project that I don't even use.

If anyone wants to see the bugs in Swig fixed, and will share my vision set forth in the [contributing guidelines](https://github.com/paularmstrong/swig/blob/master/CONTRIBUTING.md) (ex: jslint, test-driven-development, code coverage, etc), please feel free to step up and help be a maintainer of the project. To do so, take a look at the open issues and fix a bunch through some quality, lint-free, test-covered pull requests.

However, if you're fed up and looking for a similar alternative to Swig, I suggest trying out [Mozilla's nunjucks](https://mozilla.github.io/nunjucks/)

**Comment by mikeerickson** [comment]
Well, while that surely sucks!  Guess it is time to take Paul's advise and move over to nunjucks (which is really quite nice)

**Comment by igo** [comment]
Sad to see you go. Please consider the continuing on [IO Stock - Marketplace for Developers](http://iostock.com)

**Comment by eddywashere** [comment]
Thanks for the time you put into such an awesome project!

I was recently reading the nunjucks 2.0 issue https://github.com/mozilla/nunjucks/issues/347#issue-53800646

> I'm going to start a new effort and call it "nunjucks 2.0". It's 2.0 only because it sounds bigger, and will keep me motivated to push it though.

Maybe nunjucks 2.0 could take some developer ux tips from swig and makes things easier to register, include a date filter by default & allow logic tags like `&&` and `||`. Those are the big things holding me bacl. I'll make that suggestion over there. I think these communities would benefit from one project, with a migration strategy for swig. Thanks again.

**Comment by danilodeveloper** [comment]
I'm very sad about that... So now i'm going to migrate about 80 .html from Swig.js to any thing :cry: .

Anyway, thank you for create the Swig.js and good luck in your next challenge. See ya!

**Comment by ahaasler** [comment]
I was trying to decide swig vs nunjucks, this settles it.

**Comment by cdaringe** [comment]
swig rocks.  I used it a bunch!  thanks for the hard work

**Comment by mrazvan92** [comment]
good luck :-1:

**Comment by wraithan** [comment]
Used it recently in a project for work. Thanks a ton for building swig, it works beautifully for me. Would be nice if you put something prominent in your docs about the maintenance status. I hadn't jumped to the GitHub page before I started using it since the docs were the first google hit as well as most comparison sites linking to the docs directly instead of the GitHub project.

To help with the negative impact of saying it is unmaintained, you could refer to it as seeking maintainers to help further the project or something.

Thanks again!

**Comment by corbanb** [comment]
# Demo Nunjucks setup
https://github.com/corbanb/nunjucks-express/blob/master/index.js

If anyone is using swig moving to [nunjucks](https://mozilla.github.io/nunjucks/) is going to be very easy. 

Thanks for the heads up @paularmstrong.

**Comment by bricss** [comment]
I guess, I wanna maintain it.

**Comment by atian25** [comment]
had try nunjucks, got some problem:

- custom tag is powerful , but less docs.
- python syntax , not javascript , such as `{% if list and list.length%} ` , not support `{% if list && list.length%}` , and it will never change.
- maintain is low activity now

found https://github.com/justjohn/twig.js , but still unstable.

**Comment by atian25** [comment]
@bricss great to hear that.

**Comment by bricss** [comment]
I assume we need to create some sort of community or working group and get somehow control on this repo to continue dev process.
In other case maybe some organization or foundation could take care of it.

**Comment by bricss** [comment]
Paul, are you alive?
Have any ideas?

**Comment by webxl** [comment]
@wraithan +1 and @atian25 +1 Running into issues all over the place with nunjucks and metalsmith.

Who wants to fork?

**Comment by bricss** [comment]
Indeed, I wanna fork.

**Comment by cdaringe** [comment]
maybe we should set up an org for it. swig/swig, then add a few collabs?

**Comment by mikeerickson** [comment]
I am down with the idea and would surely be interested in helping

**Comment by cdaringe** [comment]
the swig org is already a thing. i propose that we:

- choose a diff org name
- fork
- put interested parties in as collaborators
- use the "it takes two +1s from collabs" to land a PR into master

**Comment by bricss** [comment]
Does anybody has an idea of fancy name for fork and domain name?

**Comment by mrazvan92** [comment]
I'm in :+1: I will contribute 

@bricss what do you thing about gisy template, escada template, walkoff, apex :dancer:

**Comment by cdaringe** [comment]
probably shouldn't change the project name, just a new org name.  project name has deep roots in the src, site, npm, etc etc

**Comment by mikeerickson** [comment]
I totally agree, keep the name. It is still a valuable project, just need some fresh eyes and fingers

**Comment by cdaringe** [comment]
could keep it simple.  `node-swig`?

**Comment by rdiaztushman** [comment]
Agreed. Same name

--
Rafael Diaz-Tushman, President & CEO
Dioscouri Design: Form and Function
www.dioscouri.com
https://www.linkedin.com/in/rdiaztushman 
w. 917-830-7411

**Comment by cdaringe** [comment]
ok, it's a thing.  let's move the discussion over there.

repo: https://github.com/node-swig/swig
discussion: https://github.com/node-swig/swig/issues/1

thanks again @paularmstrong!  at some point, those of us interested in keeping the project alive will request npm access.  it doesn't need to be now, but is that something you can set us up with?

**Comment by cdaringe** [comment]
hey @paularmstrong, can please briefly pitch in over at https://github.com/node-swig/swig/issues/7 ?  much appreciated!

**Comment by cydrobolt** [comment]
I would be super interested in helping. I use Swig all the time and would really be disappointed if the project went to rot. How can I get involved?

**Comment by cdaringe** [comment]
Drop a couple PRs over there and well get you set up

**Comment by Markkos89** [comment]
so sad news :( I use swig a lot and I found it very usefull. thanks @paularmstrong for the hardwork.

**Comment by SusanthCom** [comment]
Asm interesting implementation. Actively developed https://github.com/twigjs/twig.js

**Comment by olets** [comment]
twig.js is a little different: it's a js port of php https://github.com/twigphp/Twig. Like Swig (and [Nunjucks](https://github.com/mozilla/nunjucks), already mentioned in this thread) Twig is in the Jinja-Django family of templating languages.

Speaking as a Twig user, more collaborators in twig.js would of course be welcomed but we definitely don't want to poach anyone :) It's great to see https://github.com/node-swig/swig-templates is going strong! 👍

**Comment by SusanthCom** [comment]
@olets  Thanks for this [link](https://github.com/node-swig/swig-templates) !!! 
Nice to see, this still active !!

@paularmstrong Special thanks for your hard work.
